### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by insacuri**
## What? and Why?
Update dependencies to more up to date ones

## Checklist
  - [x] CHANGELOG.md updated? (if required)

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def test_rsi_metadata(self):
                 self.assertValid(survey)
 
     def test_missing_version(self):
-        survey = json.loads(self.message['0.0.1'])
+        json.loads(self.message['0.0.1'])
```
Comment (jonnyshaw89) [inline tests/test_service.py]: I don't get what this test is testing?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/test_service.py (hunk 1 above)
Comment (dcdarrell9) [reply tests/test_service.py]: The test looks a bit misleading, If I'm right, the `ValidSurveyId` function checks the version number, by what's passed through `def ValidSurveyId(value, version=None):` and this json.loads is not actually getting used.
Might be worth having a look at something like...
```
def test_invalid_version(self):
    survey = json.loads(self.message['5'])
    with self.assertRaises(AttributeError):
        server.ValidSurveyId(survey['survey_id'], survey['version'])

def test_valid_version(self):
    survey = json.loads(self.message['0.0.1'])
    server.ValidSurveyId(survey['survey_id'], survey['version'])
    self.assertTrue?
```

File: test_requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
-flake8==3.2.1
+flake8==3.4.1
```
Comment (dcdarrell9) [inline test_requirements.txt]: Why can't we have these as just...
flake8
mock
six
pytest
pytest-cov
pytest-html

**Review by dcdarrell9 — CHANGES_REQUESTED** [review]

File: test_requirements.txt (hunk 2 above)
Comment (insacuri) [reply test_requirements.txt]: Sounds good.  Forgot that we can do this with test files

**Review by dcdarrell9 — APPROVED** [review]

File: tests/test_service.py [hunk 3]
```diff
@@ -168,10 +168,21 @@ def test_rsi_metadata(self):
...
         survey = json.loads(self.message['0.0.1'])
         with self.assertRaises(AttributeError):
-            server.ValidSurveyId('123')
+            server.ValidSurveyId(survey['survey_id'])
+
+    def test_valid_version(self):
+        survey = json.loads(self.message['0.0.1'])
+        value = server.ValidSurveyId(survey['survey_id'], survey['version'])
+        # ValidSurveyId doesn't have a return value, so it returns None
+        self.assertIsNone(value)
```
Comment (jonnyshaw89) [inline tests/test_service.py]: I don't think we need this assert, as all its testing is that a function with no return returns `None` which is standard python behaviour.
The test will be fine in that if it doesn't work there will be an uncaught exception.

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: tests/test_service.py (hunk 3 above)
Comment (insacuri) [reply tests/test_service.py]: Ah, fair enough. I wasn't sure if it was okay for there to be a test that didn't have an assert.

**Review by jonnyshaw89 — APPROVED** [review]
